.github/workflows/unit-test.yml

```diff
@@ -97,61 +107,129 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: '${{ inputs.python_version }}'
 
       - name: Upgrade PIP
+        if: ${{ steps.continue.outputs.confirm == 'true' }}
         run: pip install --upgrade pip && pip install wheel --upgrade
 
       - name: Install PennyLane dependencies
+        if: ${{ steps.continue.outputs.confirm == 'true' }}
         run: |
           pip install -r requirements-ci.txt --upgrade
           pip install -r requirements-dev.txt --upgrade
 
       - name: Install PyTorch
-        if: inputs.install_pytorch == true
+        if: ${{ (inputs.install_pytorch == true) && (steps.continue.outputs.confirm == 'true') }}
         env:
           TORCH_VERSION: ${{ inputs.pytorch_version != '' && format('=={0}', inputs.pytorch_version) || '' }}
         run: pip install "torch${{ env.TORCH_VERSION }}" -f https://download.pytorch.org/whl/torch_stable.html
 
       - name: Install TensorFlow
-        if: inputs.install_tensorflow == true
+        if: ${{ (inputs.install_tensorflow == true) && (steps.continue.outputs.confirm == 'true') }}
         env:
           TF_VERSION: ${{ inputs.tensorflow_version != '' && format('~={0}', inputs.tensorflow_version) || '' }}
         run: pip install "tensorflow${{ env.TF_VERSION }}" "keras${{ env.TF_VERSION }}"
 
       - name: Install JAX
-        if: inputs.install_jax == true
+        if: ${{ (inputs.install_jax == true) && (steps.continue.outputs.confirm == 'true') }}
         env:
           JAX_VERSION: ${{ inputs.jax_version != '' && format('=={0}', inputs.jax_version) || '' }}
         run: pip install "jax${{ env.JAX_VERSION}}" "jaxlib${{ env.JAX_VERSION }}"
 
       - name: Install additional PIP packages
-        if: inputs.additional_pip_packages != ''
+        if: ${{ (inputs.additional_pip_packages != '') && (steps.continue.outputs.confirm == 'true') }}
         run: pip install ${{ inputs.additional_pip_packages }}
 
       - name: Install PennyLane
+        if: ${{ steps.continue.outputs.confirm == 'true' }}
         run: |
           python setup.py bdist_wheel
           pip install dist/PennyLane*.whl
 
       - name: Install PennyLane-Lightning master
-        if: inputs.install_pennylane_lightning_master == true
+        if: ${{ (inputs.install_pennylane_lightning_master == true) && (steps.continue.outputs.confirm == 'true') }}
         run: pip install -i https://test.pypi.org/simple/ PennyLane-Lightning --pre --upgrade
 
       - name: Run PennyLane Unit Tests
+        if: ${{ inputs.pipeline_mode == 'unit-tests' }}
         env:
           PYTEST_MARKER: ${{ inputs.pytest_markers != '' && format('-m "{0}"', inputs.pytest_markers) || '' }}
         # Calling PyTest by invoking Python first as that adds the current directory to sys.path
         run: python -m pytest  ${{ inputs.pytest_test_directory }} ${{ inputs.pytest_coverage_flags }} ${{ inputs.pytest_additional_args }} ${{ env.PYTEST_MARKER }} -n auto
```

Comment:
We can likely merge this step with `Run PennyLane Benchmarks` step, let's sync offline on this

Reply:
Sure! Talk with you on Monday.